src/Microsoft.ComponentDetection.Detectors/gradle/GradleComponentDetector.cs

```diff
@@ -87,4 +108,38 @@ private MavenComponent CreateMavenComponentFromFileLine(string line)
     }
 
     private bool StartsWithLetter(string input) => StartsWithLetterRegex.IsMatch(input);
+
+    private bool IsDevDependencyByConfigurations(string line)
+    {
+        if (this.prodDepsRegex == null && this.devDepsRegex == null)
+        {
+            return false; // no regexes configured to check against
+        }
+
+        var equalsSeparatorIndex = line.IndexOf('=');
+        if (equalsSeparatorIndex == -1)
+        {
+            // We can't parse out the configuration. Maybe the project is using the one-lockfile-per-configuration format but
+            // this is deprecated in Gradle so we don't support it here, projects should upgrade to one-lockfile-per-project.
```

Comment:
can we log a warning here for customers with a recommendation?

Reply:
I considered this, but for any projects that are still using one-lockfile-per-configuration, the warning would be horribly verbose as this function is executed a lot.